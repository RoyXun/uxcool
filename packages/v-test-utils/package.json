{
  "name": "@suning/v-test-utils",
  "version": "1.0.10",
  "main": "src/index.js",
  "module": "src/index.js",
  "license": "MIT",
  "scripts": {
    "start": "npm run server",
    "compile": "v-tools run compile",
    "server": "v-tools run server"
  },
  "peerDependencies": {
    "vue": ">= 2.5.*"
  },
  "devDependencies": {
    "@suning/v-tools": "^1.0.35",
    "@vue/test-utils": "^1.0.0-beta.16"
  },
  "dependencies": {
    "@suning/v-trigger": "^1.0.64",
    "@vue/server-test-utils": "^1.0.0-beta.30",
    "jquery": "^3.3.1",
    "vue-server-renderer": "^2.6.11"
  }
}
