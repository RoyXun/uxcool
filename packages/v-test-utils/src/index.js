import Vue from 'vue';

import $ from 'jquery';
import { mount, createWrapper } from '@vue/test-utils';
import Trigger from '@suning/v-trigger';

export * from '@vue/test-utils';

// export * as server from '@vue/server-test-utils';

export { $, $ as jQuery };
// ms
export async function waitTime(delay = 20) {
  await new Promise((resolve) => {
    setTimeout(() => {
      resolve();
    }, delay);
  });
}

export function removePopup() {
  document.body.innerHTML = '';
}

const vueInst = new Vue();
export function getVueCreateElement() {
  return vueInst.$createElement;
}

/**
 * 获取Trigger实例
 * @param {Wrapper} wrapper
 */
export async function getPortal(wrapper) {
  const trigger = wrapper.find(Trigger);
  const portal = createWrapper(trigger.vm.portal.$el);
  return portal;
}

/**
 * 触发事件
 * @param {Wrapper} wrapper
 * @param {string} eventName
 * @param {number} delay
 */
export async function triggerEvent(wrapper, eventName, delay = 20) {
  wrapper.trigger(eventName);
  await waitTime(delay);
}

export function mountPickerFactory(Component) {
  return async (opts = {}) => {
    const wrapper = mount(Component, {
      ...opts,
    });
    await waitTime();
    return wrapper;
  };
}

export function destroyWrapper(wrapper) {
  wrapper.destroy();
}
