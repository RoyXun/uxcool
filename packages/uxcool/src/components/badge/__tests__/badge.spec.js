import { mountPickerFactory, waitTime } from '@suning/v-test-utils';
import UxBadge from '../index';

const mountBadge = mountPickerFactory(UxBadge);
describe('badge', () => {
  it('render correctly', async () => {
    const wrapper = await mountBadge({
      propsData: {
        count: 9,
      },
      slots: {
        default: '<div style="width: 60px; height: 60px;"></div>',
      },
    });
    expect(wrapper.element).toMatchSnapshot();
  });

  it('render correctly on overflowed', async () => {
    const wrapper = await mountBadge({
      propsData: {
        overflowCount: 99,
        count: 99,
      },
    });
    const innerElement = wrapper.find('.ux-badge-count').find('span');
    expect(innerElement.element.innerHTML).toBe('99');
    wrapper.setProps({ count: 100 });
    await waitTime();
    expect(innerElement.element.innerHTML).toBe('99+');
  });

  it('render dot correctly', async () => {
    const wrapper = await mountBadge({
      propsData: {
        dot: true,
      },
    });
    expect(wrapper.element).toMatchSnapshot();
  });

  it('render show-zero correctly', async () => {
    const wrapper = await mountBadge({
      propsData: {
        count: 0,
      },
    });
    expect(wrapper.contains('.ux-badge-count')).not.toBeTruthy();
    wrapper.setProps({ showZero: true });
    await waitTime();
    expect(wrapper.contains('.ux-badge-count')).toBeTruthy();
    expect(wrapper.find('.ux-badge-count').find('span').element.innerHTML).toBe('0');
  });

  it('render text correctly', async () => {
    const wrapper = await mountBadge({
      propsData: {
        text: 'test text',
        dot: true,
      },
    });
    expect(wrapper.find('.ux-badge-status-text').element.innerHTML).toBe('test text');
  });

  it.each(['success', 'error', 'default', 'processing', 'warning'])(
    'render status correctly',
    async (status) => {
      const wrapper = await mountBadge({
        propsData: {
          status,
        },
      });
      expect(wrapper.element).toMatchSnapshot();
    }
  );
});
