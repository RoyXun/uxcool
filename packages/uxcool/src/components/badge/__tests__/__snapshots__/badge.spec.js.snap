// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`badge render correctly 1`] = `
<span
  class="ux-badge ux-badge-light"
>
  <div
    style="width: 60px; height: 60px;"
  />
  <sup
    class="ux-scroll-number ux-badge-count ux-badge-zoom-enter ux-badge-zoom-enter-active"
  >
    <span
      class="ux-scroll-number-only"
    >
      9
    </span>
  </sup>
</span>
`;

exports[`badge render dot correctly 1`] = `
<span
  class="ux-badge ux-badge-light ux-badge-not-a-wrapper"
>
  <sup
    class="ux-scroll-number ux-badge-dot"
  >
    <span
      class="ux-scroll-number-only"
    />
  </sup>
</span>
`;

exports[`badge render status correctly 1`] = `
<span
  class="ux-badge ux-badge-light ux-badge-status"
>
  <span
    class="ux-badge-status-dot ux-badge-status-success"
  />
  <span
    class="ux-badge-status-text"
  />
</span>
`;

exports[`badge render status correctly 2`] = `
<span
  class="ux-badge ux-badge-light ux-badge-status"
>
  <span
    class="ux-badge-status-dot ux-badge-status-error"
  />
  <span
    class="ux-badge-status-text"
  />
</span>
`;

exports[`badge render status correctly 3`] = `
<span
  class="ux-badge ux-badge-light ux-badge-status"
>
  <span
    class="ux-badge-status-dot ux-badge-status-default"
  />
  <span
    class="ux-badge-status-text"
  />
</span>
`;

exports[`badge render status correctly 4`] = `
<span
  class="ux-badge ux-badge-light ux-badge-status"
>
  <span
    class="ux-badge-status-dot ux-badge-status-processing"
  />
  <span
    class="ux-badge-status-text"
  />
</span>
`;

exports[`badge render status correctly 5`] = `
<span
  class="ux-badge ux-badge-light ux-badge-status"
>
  <span
    class="ux-badge-status-dot ux-badge-status-warning"
  />
  <span
    class="ux-badge-status-text"
  />
</span>
`;
